js/components/generation/select-sources-btn.js

```diff
@@ -23,17 +23,16 @@ define([
 			this.label = typeof params.label === 'undefined' ? 'Generate' : params.label;
 			this.wasGenerated = typeof params.wasGenerated === 'undefined' ? false : params.wasGenerated;
 
-			this.sources = ko.computed(() => params.sources().map(s => ({
+			this.sourceOptions = ko.computed(() => (ko.utils.unwrapObservable(params.sources) || []).map(s => ({
```

Comment:
If we pass sources to the popup anyway, do we really need this `map`? - it actually doesn't do anything useful then

Reply:
From what I recall, the map is projecting a name and disabled field from the underlying source object, so the source is becoming something new (which is why I added the 'source' field to keep track of the origional, and called this collecton 'sourceOptions' instead of leaving it as 'source' because the param `source` and the mapped `source` were different things). But if the source object can be passed in 'as-is' and we modify the popup logic to read values directly from the source object, then we may not need this map.  However, this also is here to handle the case where the component did not receive an array of sources at all (you can make it work with just passing 'selectedSource').  

So if you want to try to remove this `map()`, where do you see the impact on subsequent code (the popup, the toggles, etc)?

Reply:
> where do you see the impact on subsequent code (the popup, the toggles, etc)?

As you truly mentioned - only in those places which rely on `name` (so we'll have to switch those to `sourceName`) and `disabled` (move the undefined check) + update your new code from using nested `sourceOption.source` to the `source`. But this will allow to remove a level of data nesting and processing

Reply:
Ok, let me work on it.